gov2/bedrock-runtime/cmd/main.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/awsdocs/aws-doc-sdk-examples/gov2/demotools"
+	"github.com/awsdocs/aws-doc-sdk-examples/gov2/bedrock-runtime/scenarios"
+)
+
+// main loads default AWS credentials and configuration from the ~/.aws folder and runs
+// a scenario specified by the `-scenario` flag.
+//
+// `-scenario` can be one of the following:
+//
+// * `invokemodels`  -	Runs a scenario that shows how to invoke various image and text
+//						generation models on Amazon Bedrock.
+func main() {
+	scenarioMap := map[string]func(sdkConfig aws.Config){
+		"invokemodels":    runInvokeModelsScenario,
+	}
+	choices := make([]string, len(scenarioMap))
+	choiceIndex := 0
+	for choice := range scenarioMap {
+		choices[choiceIndex] = choice
+		choiceIndex++
+	}
+	scenario := flag.String(
+		"scenario", "",
+		fmt.Sprintf("The scenario to run. Must be one of %v.", choices))
+	flag.Parse()
+
+	if runScenario, ok := scenarioMap[*scenario]; !ok {
+		fmt.Printf("'%v' is not a valid scenario.\n", *scenario)
+		flag.Usage()
+	} else {
+		sdkConfig, err := config.LoadDefaultConfig(context.Background(), config.WithRegion("us-east-1"))
```

Comment:
Either don't specify region (i.e., take whatever is configured in .aws/config) or make it an input. I'm in `us-west-2` and it took me a while to discover why I kept getting 403 errors.

Reply:
The issue with Bedrock is that the available models are different throughout the regions, and `us-east-1` is the only one where all the models used in the examples are available. I agree it's a bit of a hack, but I'd rather keep it hard-coded until there is consistency across regions. I can add it to the README to make it clearer. What do you think?

Reply:
I've introduced the ability to configure the AWS region via a new `-region` flag in both `./hello/hello.go` and `./cmd/main.go`. While the default region remains `us-east-1`, users now have the flexibility to specify an alternative region that suits their needs.

Key Changes:
- Added `-region` flag to `./hello/hello.go` and `./cmd/main.go`.
- Ensured the default AWS region is set to `us-east-1`.
- Updated Readme with a new **Region Configuration** section to guide users on how to set the region flag.
- Enhanced error handling in examples to provide clearer guidance when the service or model is not available in the specified region.

The modifications are intended to offer more control to users over the region settings and to improve the usability of the examples by addressing potential regional availability issues.

Please review the updates and let me know if there are any concerns or further enhancements needed.